lib/shopify-cli/transform_data_structure.rb

```diff
@@ -0,0 +1,77 @@
+
+module ShopifyCli
+  ##
+  # `TransformDataStructure` helps to standardize data structure access. It
+  # traverses nested data structures and can convert
+  #
+  # * all strings used as keys to symbols,
+  # * all strings used as keys from `CamelCase` to `snake_case` and
+  # * associative array containers, e.g. from Hash to OpenStruct.
+  #
+  # Standardizing how a data structure is accessed can greatly reduces the risk
+  # of subtle bugs especially when dealing with API responses.
+  #
+  #    TransformDataStructure.new(
+  #      symbolize_keys: true,
+  #      underscore_keys: true,
+  #      associative_array_container: OpenStruct
+  #    ).call([{"SomeKey" => "value"}]).tap do |result|
+  #      result.value # => [#<OpenStruct: some_key: "value">]
+  #    end
+  #
+  # Since `TransformDataStructure` is a method object, it can easily be chained:
+  #
+  #    require 'open-uri'
+  #    ShopifyCli::Result
+  #      .call { open("https://jsonplaceholder.typicode.com/todos/1") }
+  #      .map(&TransformDataStructure.new(symbolize_keys: true, underscore_keys; true))
+  #      .value # => { id: 1, user_id: 1, ... }
+  #
+  class TransformDataStructure
+    include ShopifyCli::MethodObject
+
+    property! :underscore_keys, accepts: [true, false], default: false, reader: :underscore_keys?
+    property! :symbolize_keys, accepts: [true, false], default: false, reader: :symbolize_keys?
+    property! :associative_array_container, accepts: Class, default: Hash
```

Comment:
🙋‍♀️ could there be any issues with using `Class` here instead of something more defined, like `OpenStruct` since that is the default? i.e. any possible dangers of someone passing a `Class` that is not what we'd expect?